src/main/scala/PropertyChecker.scala

```diff
@@ -86,174 +100,485 @@ object PropertyChecker {
     "reference" -> referenceProperty(PropertyType.ForeignKey),
     // foreignKey reference properties
     "resource" -> resourceProperty(PropertyType.ForeignKeyReference),
-    "schemaReference" -> schemaReferenceProperty(PropertyType.ForeignKeyReference))
+    "schemaReference" -> schemaReferenceProperty(
+      PropertyType.ForeignKeyReference
+    )
+  )
 
+  def checkProperty(
+      property: String,
+      value: JsonNode,
+      baseUrl: String,
+      lang: String
+  ): (JsonNode, Array[String], PropertyType.Value) = {
+    val propertyRegEx = "^[a-z]+:".r
+    val matcher = propertyRegEx.pattern.matcher(property)
+    if (Properties.contains(property)) {
+      val f = Properties(property)
+      f(value, baseUrl, lang)
+    } else if (
+      matcher.matches() && NameSpaces.values.contains(property.split(":")(0))
+    ) {
+      val (newValue, warnings) = checkCommonPropertyValue(value, baseUrl, lang)
+      (newValue, warnings, PropertyType.Annotation)
+    } else {
+      // property name must be an absolute URI
+      try {
+        val scheme = new URI(property).getScheme
+        if (scheme.isEmpty)
+          return (
+            value,
+            Array[String]("invalid_property"),
+            PropertyType.Undefined
+          )
+        val (newValue, warnings) =
+          checkCommonPropertyValue(value, baseUrl, lang)
+        (newValue, warnings, PropertyType.Annotation)
+      } catch {
+        case e: Exception => {
+          (value, Array[String]("invalid_property"), PropertyType.Undefined)
+        }
+      }
+    }
+  }
 
-  def checkProperty(property: String, value: JsonNode, baseUrl:String, lang:String): (JsonNode, Array[String], PropertyType.Value) = {
-    // More conditions and logic to add here.
-    val f = Properties(property)
-    return f(value, baseUrl, lang)
+  def checkCommonPropertyValue(
+      value: JsonNode,
+      baseUrl: String,
+      lang: String
+  ): (JsonNode, Array[String]) = {
+    value match {
+      case o: ObjectNode => {
+        val valueCopy = o.deepCopy()
+        var warnings = Array[String]()
+        val fieldsAndValues = Array.from(valueCopy.fields().asScala)
+        for (fieldAndValue <- fieldsAndValues) {
+          val p = fieldAndValue.getKey
+          var v = fieldAndValue.getValue
+          p match {
+            case "@context" =>
```

Comment:
Could simplify this to `case "@context" | "@list" | "@set" => throw new MetadataError(s"$p: common propetty has $p property"`